### ISSUE TITLE
Harmonize the description of several standard names for various types of precipitation

### ISSUE THREAD

**Issue by larsbarring** [cf-convention/vocabularies#51]
Some time ago, when looking at cf-convention/discuss#159, I discovered inconsistent and/or rather too brief description(s) of several standard names for various types precipitation. Hence I think that it would be useful to make a concerted effort to harmonise these. 

problem | standard name | description | canonical unit
-- | -- | -- | --
too brief| convective_rainfall_amount | "Amount" means mass per unit area. | kg m-2
too brief | convective_rainfall_flux | In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics. | kg m-2 s-1
too brief | convective_rainfall_rate | _No help available_ | m s-1
too brief | rainfall_amount | "Amount" means mass per unit area. | kg m-2
too brief | rainfall_flux | In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics. | kg m-2 s-1
too brief | rainfall_rate | _No help available_ | m s-1
term mixup | stratiform`_graupel_` flux </br> Alias:  </br> large_scale`_graupel_` flux | In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics. Stratiform `precipitation, whether liquid or frozen, is precipitation` that formed in stratiform cloud. Graupel consists of heavily rimed snow particles, often called snow pellets; often indistinguishable from very small soft hail except when the size convention that hail must have a diameter greater than 5 mm is adopted. Reference: American Meteorological Society Glossary http://glossary.ametsoc.org/wiki/Graupel. There are also separate standard names for hail. Standard names for "graupel_and_hail" should be used to describe data produced by models that do not distinguish between hail and graupel. | kg m-2 s-1
term mixup (#159) | stratiform`_rainfall_`amount  </br> Alias:  </br> large_scale`_rainfall_`amount | Stratiform `precipitation`, whether liquid or frozen, is `precipitation` that formed in stratiform cloud. "Amount" means mass per unit area. | kg m-2
term mixup | stratiform`_rainfall_`flux  </br> Alias:  </br> large_scale``rainfall_`flux | In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics. Stratiform `precipitation`, whether liquid or frozen, is `precipitation` that formed in stratiform cloud. | kg m-2 s-1
term mixup | stratiform`_rainfall_`rate  </br> Alias:  </br> large_scale`_rainfall_`rate | Stratiform `precipitation`, whether liquid or frozen, is `precipitation` that formed in stratiform cloud. | m s-1
term mixup| stratiform`_snowfall_`amount  </br> Alias:  </br> large_scale`_snowfall_`amount | Stratiform `precipitation`, whether liquid or frozen, is `precipitation` that formed in stratiform cloud. "Amount" means mass per unit area. | kg m-2
term mixup | stratiform`_snowfall_`flux  </br> Alias:  </br> large_scale`_snowfall_`flux | In accordance with common usage in geophysical disciplines, "flux" implies per unit area, called "flux density" in physics. Stratiform `precipitation`, whether liquid or frozen, is `precipitation` that formed in stratiform cloud. | kg m-2 s-1

This table only include standard names where I think that there is room for improvements and/or clarifications. Related to these 12 ones there are 43 other standard names from which useful boilerplate pieces of text can be extracted, or otherwise used as inspiration. Thus I believe the work involved is not overwhelming to improve the description of these 12 standard names. 

In addition to these standard names in need of better descriptions, there is one standard name that I believe is wrong or at least inconsistent:

standard name | description | canonical unit
-- | --| --
convective_precipitation_rate | "Precipitation rate" means the depth or thickness of the layer formed by precipitation per unit time. Convective precipitation is that produced by the convection schemes in an atmosphere model. "Precipitation" in the earth's atmosphere means precipitation of water in all phases. | m s-1 | convective_precipitation_rate is the sum of convective_rainfall_rate and what (and the three non-existing convective_snow_\|  _hail_ \| _graupel_fall_rate)? This would be inconsistent because of what the height represents. This quantity is better represented by the already existing lwe_convective_precipitation_rate

The problem with this is that `convective_precipitation_rate` (supposedly, and in line with other standard names) is the sum of `convective_rainfall_rate` and _what_ (and the three non-existing `convective_snow_`|  `_hail_` | `_graupel_fall_rate`)? This would be inconsistent because of what the height represents. 

Thus I suggest that `convective_precipitation_rate` is deprecated in favour of `lwe_convective_precipitation_rate`.

The full table of the standard names is attached as an .xls spreadsheet file:
[Precip_Standard_Names_gh.xlsx](https://github.com/cf-convention/discuss/files/12677434/Precip_Standard_Names_gh.xlsx)

**Comment by efisher008** [comment]
Dear Lars @larsbarring,

I am revisiting this issue while looking through the currently open issues in the new **cf-conventions/vocabularies** repo. I realise this was created last year and unfortunately never progressed beyond the CF workshop 2023. Do you still feel it's worth pursuing? Perhaps we could revive the conversation in time for the CF workshop 2024 this September. 

Best regards,
Ellie

**Comment by larsbarring** [comment]
Dear Ellie @efisher008,

I have not been thinking about this for quite some time. But in general I think that all the different variants of precipitation standard names -- due to the difference in physical quantities and their units, in combination with the various types of hydrometeors -- should be carefully crafted and explained/defined to help users unambiguously find the correct standard name for their data.

But since back then  I come to understand that much of what is in the first table is related to the term database of the cf-editor. As I have no insight into this I am not sure that I can add much, except help out with comments and suggestions if someone would be willing and have time to take this on.

Regarding `convective_precipitation_rate`, which i believe should be deprecated, it might be handled as a separate issue.

Cheers,
Lars

**Comment by efisher008** [comment]
Hi Lars @larsbarring,

Thank you for your reply. I agree that it would be valuable to clarify the description, and in some cases, restructure, these precipitation standard names to ensure there is a) no overlap in meaning b), no ambiguity in their interpretation, and c) ease of findability for the correct name to use.

I have commented in #67 that the subject of that issue (clarifying the description of the term `rain`/`rainfall` in various standard names, especially that of `stratiform_rainfall_rate`) could be transferred to the prospective changes/improvements being proposed in this issue. We shall see what the decision is there. 

To be clear, when you refer to the "[term database of the cf-editor](https://github.com/cf-convention/vocabularies/issues/51#issuecomment-2270977316)", are you talking about the CF phrasebank, where terms used in standard name descriptions are defined for future use? It could be worth updating some descriptions to make terms clearer/less ambiguous if this is desired.

Best wishes,
Ellie

**Comment by taylor13** [comment]
I am not sure this is backward-compatible, but I think "rain" should be reserved for liquid water only and "frozen water" should refer to all non-liquid droplets (snow, hail, etc.). "precipitation" would be the sum of falling rain+frozen water.  

We could also consider distinguishing between "rainfall" and "rain", with the former reserved to references to the rain at the surface, but I don't know what the comparable terms would be for precipitation and frozen water.

**Comment by larsbarring** [comment]
Yes, I agree with you first point, which fits nicely with #59. Regarding your second point, isn't there both "snowfall" and "hailfall" that could be used in analogy with "rainfall"? Or do the latter two more clearly (than "rainfall") relate to the process of *falling* rather than accumulation on the ground. I never heard of "graupelfall", although I imagine that we could create the term if ever needed.

**Comment by JonathanGregory** [comment]
I agree with Lars that `convective_precipitation_rate` ought to be deprecated, because of the mixture of phase. `lwe_convective_precipitation_rate` is much to be preferred.

I agree with Karl that `rain` is liquid water and `frozen_water` is all forms of solid water. I devised some of these standard names, a long time ago, and tried to be consistent in using `rainfall` for rain when thinking of it as arriving at the surface, with `rain` for thinking of it as constituent of the air. There is a similar distinction between `snowfall` and `snow`, and also a distinction between `snowfall` and `surface_snow`. The latter is useful because "snow amount" can refer to either (falling or lying).

**Comment by JonathanGregory** [comment]
We haven't made a similar distinction for `precipitation`. For example, `mass_fraction_of_precipitation_in_air` refers to particles of condensed water of any form that are in the air, and on the way down, I suppose, while `mass_fraction_of_solid_precipitation_falling_onto_surface_snow` refers to precipitation arriving at the ground. I think we don't strictly _need_ the distinction between `rain` and `rainfall`, but since English has this distinction, we can try to use those words consistently.

**Comment by taylor13** [comment]
I've made a relevant comment here: https://github.com/cf-convention/vocabularies/issues/67#issuecomment-2278882455
